src/InvariantTheory/invariant_rings.jl

```diff
@@ -127,16 +127,81 @@ function Base.show(io::IO, IR::InvRing)
 end
 
 function reynolds_molien_via_singular(IR::InvRing)
+   @assert !ismodular(IR)
    if !isdefined(IR, :reynolds_singular) || !isdefined(IR, :molien_singular)
       R = polynomial_ring(IR)
       singular_matrices = _action_singular(IR)
-      rey, mol = Singular.LibFinvar.reynolds_molien(singular_matrices...)
+
+      if iszero(characteristic(coefficient_ring(IR)))
+         rey, mol = Singular.LibFinvar.reynolds_molien(singular_matrices...)
+      else
+         error("Not implemented yet") # See https://github.com/oscar-system/Singular.jl/issues/486
+      end
       IR.reynolds_singular = rey
       IR.molien_singular = mol
    end
    return IR.reynolds_singular, IR.molien_singular
 end
 
+function reynolds_operator(IR::InvRing{FldT, GrpT, T}, f::T) where {FldT, GrpT, T <: MPolyElem}
+   @assert parent(f) === polynomial_ring(IR)
+
+   rey, _ = reynolds_molien_via_singular(IR)
+   fSing = singular_ring(polynomial_ring(IR))(f)
+   fReySing = Singular.LibFinvar.evaluate_reynolds(rey, fSing)
+   # fReySing is an ideal...
+   @assert length(gens(fReySing)) == 1
+   return polynomial_ring(IR)(gens(fReySing)[1])
+end
+
+function molien_series(IR::InvRing)
```

Comment:
For functions like `minpoly`, it is convention to supply the univariate polynomial ring in which the result should lie as first argument. Perhaps follow that here, too?

Mind you, I'd be happy to have it be an *optional* argument, for convenience when using it interactive.